integration-tests/tests/utils/docker_utils.py

```diff
@@ -0,0 +1,72 @@
+"""Provide utility functions related to the use of Docker during integration tests."""
+
+import re
+import shutil
+import subprocess
+from enum import Enum
+
+
+class DockerStatus(str, Enum):
+    """Enum of possible Docker Statuses."""
+
+    created = "created"
+    restarting = "restarting"
+    running = "running"
+    removing = "removing"
+    paused = "paused"
+    exited = "exited"
+    dead = "dead"
+
+
+def get_docker_binary_path() -> str:
+    """
+    Finds the absolute path to the Docker client in the current PATH.
+
+    :return: Absolute path to the Docker binary.
+    :raise RuntimeError: docker is not found on PATH.
+    """
+    docker_bin = shutil.which("docker")
+    if docker_bin is None:
+        err_msg = "docker not found in PATH"
+        raise RuntimeError(err_msg)
+    return docker_bin
+
+
+def list_running_containers_with_prefix(prefix: str) -> list[str]:
+    """
+    Lists running Docker containers whose names begin with `prefix` and end with one or more digits.
+
+    :param prefix:
+    :return: List of running container names that match the pattern.
+    :raise RuntimeError: Error while invoking docker ps or parsing its output.
+    """
+    docker_bin = get_docker_binary_path()
+
+    try:
+        ps_proc = subprocess.run(
+            [
+                docker_bin,
+                "ps",
+                "--format",
+                "{{.Names}}",
+                "--filter",
+                f"name={prefix}",
+                "--filter",
+                "status=running",
+            ],
+            capture_output=True,
+            text=True,
+            check=True,
+        )
+    except subprocess.CalledProcessError as ex:
+        err_out = (ex.stderr or ex.stdout or "").strip()
+        err_msg = f"Error listing running containers for prefix {prefix}: {err_out}"
+        raise RuntimeError(err_msg) from ex
```

Comment:
We do not need to capture or rewrap the subprocess output here because `pytest` already prints `stderr` and `stdout`. It is also confusing to raise `stderr` as a new exception from the original `CalledProcessError`, since it makes it look as if the `CalledProcessError` caused the docker failure, when the real sequence is the reverse.  

I also tried appending `stderr` contents to the `CalledProcessError`, but it did not work out. It is simpler to let `pytest` display `stderr` directly, since `pytest` prints it before the exception message and preserves the natural order of events.

For a clearer exception stack trace, you can try using the `pytest` option `--tb=native` or `--tb=short` on the command line or in `pytest.ini`.

Reply:
If you have a strong opinion on how subprocess errors should be handled, we can agree upon a design, make it into a utils function, and publish it via a separate PR.